js/btcmarkets.js

```diff
@@ -631,111 +664,89 @@ module.exports = class btcmarkets extends Exchange {
-        const type = (order['ordertype'] === 'Limit') ? 'limit' : 'market';
-        const timestamp = this.safeInteger (order, 'creationTime');
-        if (market === undefined) {
-            market = this.market (order['instrument'] + '/' + order['currency']);
-        }
-        let status = 'open';
-        if (order['status'] === 'Failed' || order['status'] === 'Cancelled' || order['status'] === 'Partially Cancelled' || order['status'] === 'Error') {
-            status = 'canceled';
-        } else if (order['status'] === 'Fully Matched' || order['status'] === 'Partially Matched') {
-            status = 'closed';
+        const timestamp = this.parse8601 (this.safeString (order, 'creationTime'));
+        const marketId = this.safeString (order, 'marketId');
+        const symbol = this.lookupSymbolFromMarketId (marketId);
+        let side = undefined;
+        if (this.safeString (order, 'side') === 'Bid') {
+            side = 'buy';
+        } else {
+            side = 'sell';
         }
-        const price = this.safeFloat (order, 'price') / multiplier;
-        const amount = this.safeFloat (order, 'volume') / multiplier;
-        const remaining = this.safeFloat (order, 'openVolume', 0.0) / multiplier;
+        const type = this.safeStringLower (order, 'type');
+        const price = this.safeFloat (order, 'price');
+        const amount = this.safeFloat (order, 'amount');
+        const remaining = this.safeFloat (order, 'openAmount');
         const filled = amount - remaining;
-        const trades = this.parseMyTrades (order['trades'], market);
-        const numTrades = trades.length;
-        let cost = filled * price;
-        let average = undefined;
-        let lastTradeTimestamp = undefined;
-        if (numTrades > 0) {
-            cost = 0;
-            for (let i = 0; i < numTrades; i++) {
-                const trade = trades[i];
-                cost = this.sum (cost, trade['cost']);
-            }
-            if (filled > 0) {
-                average = cost / filled;
-            }
-            lastTradeTimestamp = trades[numTrades - 1]['timestamp'];
-        }
-        const id = this.safeString (order, 'id');
-        const clientOrderId = this.safeString (order, 'clientRequestId');
+        const status = this.parseOrderStatus (this.safeString (order, 'status'));
         return {
             'info': order,
-            'id': id,
-            'clientOrderId': clientOrderId,
+            'id': this.safeString (order, 'orderId'),
+            'clientOrderId': undefined,
             'timestamp': timestamp,
             'datetime': this.iso8601 (timestamp),
-            'lastTradeTimestamp': lastTradeTimestamp,
-            'symbol': market['symbol'],
+            'lastTradeTimestamp': undefined,
+            'symbol': symbol,
             'type': type,
             'side': side,
             'price': price,
-            'cost': cost,
+            'cost': price * amount,
```

Comment:
If `price` or `amount` is undefined, it will generate a NaN in JS and an exception in Python and PHP. Also, it should use `filled` instead of `amount. 

So, we need to change `'cost': price * amount,` to:

```JavaScript
let cost = undefined;
//
// or
// let cost = this.safeFloat (order, 'some_code_field_if_any');
//
if ((price !== undefined) && (filled !== undefined)) {
    cost = price * filled;
}
```

Reply:
`cost = price * filled` ?  i didn't know this was the calc for orders, i thought it was always price * amount

Reply:
@npomfret yep, for spot orders in general, if no cost is specified by the exchange API, then we calculate it as price * filled, if `filled` is defined and `price` is defined..